### ISSUE TITLE
[BUG] Dart Enhanced Enums in Flutter 3

### ISSUE THREAD

**Issue by alex-sandri** [dart-code-checker/dart-code-metrics#848]
* **Dart code metrics version: 4.15.1**
* **Dart sdk version: 2.17.0**

**Please show your full configuration:**

<details>
<summary>Configuration</summary>

```yaml
# This is, I think, the only config portion needed

dart_code_metrics:
  rules:
    - prefer-trailing-comma
```

</details>

**What did you do? Please include the source code example causing the issue.**

```dart
enum MessageBoxType {
  error,
  neutral,
  success,
  warning;

  Color get foreground {
    switch (this) {
      case MessageBoxType.error:
        return Colors.white;
      case MessageBoxType.neutral:
        return Colors.black;
      case MessageBoxType.success:
        return Colors.white;
      case MessageBoxType.warning:
        return Colors.black;
    }
  }

  Color get background {
    switch (this) {
      case MessageBoxType.error:
        return Colors.red;
      case MessageBoxType.neutral:
        return Colors.grey.shade200;
      case MessageBoxType.success:
        return Colors.green;
      case MessageBoxType.warning:
        return Colors.yellow;
    }
  }
}
```

**What did you expect to happen?**

No errors.

**What actually happened?**

```
[app_shared]: lib\src\widgets\message_box.dart:
[app_shared]: WARNING Prefer trailing comma.
[app_shared]:         lib\src\widgets\message_box.dart:7:3
[app_shared]:         prefer-trailing-comma : https://dartcodemetrics.dev/docs/rules/common/prefer-trailing-comma
[app_shared]:
[app_shared]:
[app_shared]: pub finished with exit code 2
```

**Are you willing to submit a pull request to fix this bug?**

I'd like to but have no idea on how to fix this.

**Comment by incendial** [comment]
@alex-sandri hi, thanks for the report. It complains about ` warning;` since `;` is not an expected token here. We're discussing how to address it properly.
